spec/javascripts/actions/story_spec.js

```diff
@@ -219,4 +223,89 @@ describe('Story Actions', () => {
       expect(fakeDispatch).toHaveBeenCalledWith(Story.updateHighlight(storyId, true));
     });
   });
+
+  describe('confirmBeforeSaveIfNeeded', () => {
+    const story = storyFactory();
+
+    describe('when do not need of confirmation', () => {
+      const needConfirmation = sinon.stub().returns(false);
+      const confirm = sinon.stub();
+
+      describe('and callback.onConfirmed do not throws an error', () => {
+        it('call callback.onConfirmed', async () => {
+          const callback = { onConfirmed: sinon.stub() };
+
+          await Story.confirmBeforeSaveIfNeeded(story, confirm, needConfirmation, callback);
+          expect(callback.onConfirmed).toHaveBeenCalled();
+        });
+      });
+
+      describe('and callback.onConfirmed throws an error', () => {
+        const error = { error: "boom" };
+        const callback = { 
+          onConfirmed: sinon.stub().rejects(error), 
+          onError: sinon.stub() 
+        };
+
+        beforeEach(async () => {
+          await Story.confirmBeforeSaveIfNeeded(story, confirm, needConfirmation, callback);
+        });
+
+        it('call callback.onConfirmed', () => {
+          expect(callback.onConfirmed).toHaveBeenCalled();
+        });
+
+        it('call callback.onError', () => {
+          expect(callback.onError).toHaveBeenCalled();
+        });
+      });
+    });
+
+    describe('when need of confirmation', () => {
+      describe('and is not confirmed', () => {
+        it('do not call callback.onConfirmed', async () => {
+          const callback = { onConfirmed: sinon.stub() };
+          const needConfirmation = sinon.stub().returns(true);
+          const confirm = sinon.stub().returns(false);
+
+          await Story.confirmBeforeSaveIfNeeded(story, confirm, needConfirmation, callback);
+          expect(callback.onConfirmed).not.toHaveBeenCalled();
+        });
+      });
+
+      describe('and is confirmed', () => {
+        const needConfirmation = sinon.stub().returns(true);
+        const confirm = sinon.stub().returns(true);
+
+        describe('and callback.onConfirmed do not throws an error', () => {
+          it('call callback.onConfirmed', async () => {
+            const callback = { onConfirmed: sinon.stub() };
+
+            await Story.confirmBeforeSaveIfNeeded(story, confirm, needConfirmation, callback);
+            expect(callback.onConfirmed).toHaveBeenCalled();
+          });
+        });
+
+        describe('and callback.onConfirmed throws an error', () => {
+          const error = { error: "boom" };
+          const callback = { 
+            onConfirmed: sinon.stub().rejects(error), 
+            onError: sinon.stub() 
+          };
+
+          beforeEach(async () => {
+            await Story.confirmBeforeSaveIfNeeded(story, confirm, needConfirmation, callback);
```

Comment:
This is the "act" part, we should move it to inside the `it` and move the creation of the `callback` and `error` object to inside the `beforeEach` (since they are setup) :smile: 

Another important thing to notice here is that we're creating the stubs only once (will be fixed when moved to `beforeEach`), so we're sharing state between the tests which shouldn't happen.